test/join.jl

```diff
@@ -578,4 +578,54 @@ module TestJoin
         @test eltypes(join(l, r, on=:b, kind=:outer, makeunique=true)) ==
             [Union{Int, Missing}, CS, Union{Int, Missing}]
     end
+
+    @testset "test checks of merge key uniqueness" begin
+        @test_throws ArgumentError join(name, job, on=:ID,
+                                        validate=(false, true))
+        @test_throws ArgumentError join(name, job, on=:ID,
+                                        validate=(true, true))
+        @test_throws ArgumentError join(job, name, on=:ID,
+                                        validate=(true, false))
+        @test_throws ArgumentError join(job, name, on=:ID,
+                                        validate=(true, true))
+        @test_throws ArgumentError join(job, job, on=:ID,
+                                        validate=(true, true))
+
+
+        @test join(name, job, on=:ID, validate=(true, false)) ==  inner
+        @test join(name, job, on=:ID, kind=:inner, validate=(false, false)) == inner
+
+        # Make sure ok with various weird characters
+        for weird in [missing, NaN, 0.0, -0.0]
+            name_w_missing = DataFrame(ID = [1, 2, 3, weird],
+                                       Name = Union{String, Missing}["John Doe", "Jane Doe",
+                                                                     "Joe Blogs", "Maria Tester"])
+            @test join(name_w_missing, job, on=:ID, validate=(true, false)) ==  inner
+
+            # Make sure duplicated missings still an exception
```

Comment:
Comment and variable names should be adjusted since you test other values than missing.